### PR TITLE
Issue 261

### PR DESCRIPTION
**Pull request by ddcprg**
Internal code review

### CODE REVIEW

File: src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2016-2017 Expedia Inc.
```
Comment (massdosage) [inline src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java]: Minor, update year to 2018

File: src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Emulates a S3 endpoint.
+ */
+public class S3ProxyRule extends ExternalResource {
+  private static final Logger LOG = LoggerFactory.getLogger(S3ProxyRule.class);
```
Comment (massdosage) [inline src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java]: LOG -> logger (to match upstream project convention)

**Comment by massdosage** [comment]
Mostly LGTM, couple of small comments and some kind of usage/integration test as discussed in chat would be nice.

File: src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java (hunk 1 above)
Comment (ddcprg) [reply src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java]: done

File: src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java (hunk 2 above)
Comment (ddcprg) [reply src/main/java/org/gaul/s3proxy/junit/rules/S3ProxyRule.java]: done

**Comment by ddcprg** [comment]
Will raise again mainstream

**Pull request closed without merging**